plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/LegacyDataHandlerTests.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\DataSynchronizer;
+use Automattic\WooCommerce\RestApi\UnitTests\HPOSToggleTrait;
+use Automattic\WooCommerce\Internal\DataStores\Orders\LegacyDataHandler;
+use Automattic\WooCommerce\RestApi\UnitTests\Helpers\OrderHelper;
+
+/**
+ * Class OrdersTableQueryTests.
+ */
+class LegacyDataHandlerTests extends WC_Unit_Test_Case {
+	use HPOSToggleTrait;
+
+	/**
+	 * @var LegacyDataHandler
+	 */
+	private $sut;
+
+	/**
+	 * Initializes system under test.
+	 */
+	public function setUp(): void {
+		parent::setUp();
+
+		add_filter( 'wc_allow_changing_orders_storage_while_sync_is_pending', '__return_true' );
+		$this->setup_cot();
+
+		$this->sut = wc_get_container()->get( LegacyDataHandler::class );
+	}
+
+	/**
+	 * Destroys system under test.
+	 */
+	public function tearDown(): void {
+		$this->clean_up_cot_setup();
+		remove_all_filters( 'wc_allow_changing_orders_storage_while_sync_is_pending' );
+	}
+
+	/**
+	 * Tests the cleanup of legacy data.
+	 */
+	public function test_post_data_cleanup() {
+		$this->enable_cot_sync();
+		$orders = array(
+			OrderHelper::create_complex_data_store_order(),
+			OrderHelper::create_order(),
+		);
+		$this->disable_cot_sync();
+
+		// Confirm orders have been synced up (i.e. are not placeholders) and contain metadata.
+		foreach ( $orders as $order ) {
+			$this->assertEquals( 'shop_order', get_post_type( $order->get_id() ) );
+			$this->assertNotEmpty( get_post_meta( $order->get_id() ) );
+		}
+
+		// Check that counts are working ok.
+		$this->assertEquals( 1, $this->sut->count_orders_for_cleanup( array( $orders[0]->get_id() ) ) );
+		$this->assertEquals( 2, $this->sut->count_orders_for_cleanup() );
```

Comment:
Should we also cover ranges (perhaps we'd need at least 3 test orders to do that convincingly)? Should we cover non-happy paths like a non-existent order ID, and 'impossible' ranges like 300-2?

Reply:
Yass. I've added a bunch of additional tests (covering ranges, etc.) in 957ee272ef625bcb876515ed824b1545a8949c56.